### PR TITLE
feat: Starting point of app with Landing screens

### PR DESCRIPTION
**Pull request by ShivamMenda**
Solves #6

### CODE REVIEW

**Comment by ShivamMenda** [comment]
@chandansgowda There was a small forking problem so created another pull request with my changes.

**Comment by ShivamMenda** [comment]
https://github.com/AOSSIE-Org/Resonate/assets/74780977/4d883e58-5523-434f-8ee9-89c511120844

**Comment by chandansgowda** [comment]
@ShivamMenda Great job till here.
Can we add Resonate (text/logo) in the top left of the screens ?

**Comment by ShivamMenda** [comment]
@chandansgowda yeah will try to and send you the output.

**Comment by ShivamMenda** [comment]
https://github.com/AOSSIE-Org/Resonate/assets/74780977/d26b4a65-505a-4477-8ad0-dc362eefc760

**Comment by ShivamMenda** [comment]
@chandansgowda Added the logo as requested.

File: lib/controllers/auth_state_controller.dart [hunk 1]
```diff
@@ -87,7 +88,11 @@ class AuthStateController extends GetxController {
         Get.offNamed(AppRoutes.tabview);
       }
     } catch (e) {
-      Get.offNamed(AppRoutes.login);
+      SharedPreferences prefs = await SharedPreferences.getInstance();
+      bool? initScreen = prefs.getBool("initScreenShown");
```
Comment (chandansgowda) [inline lib/controllers/auth_state_controller.dart:92]: Rename initScreenShown to something that explains the context better. Like landingScreenShown or newUserInfoScreenShown etc

**Review by chandansgowda — CHANGES_REQUESTED** [review]
One small change. Rename the preference key name used as mentioned using a single line comment on one of the file.

File: lib/controllers/auth_state_controller.dart (hunk 1 above)
Comment (ShivamMenda) [reply lib/controllers/auth_state_controller.dart:92]: Sure.

**Comment by ShivamMenda** [comment]
@chandansgowda Changed the name and added a comment too.

File: lib/controllers/auth_state_controller.dart [hunk 2]
```diff
@@ -89,7 +89,7 @@ class AuthStateController extends GetxController {
       }
     } catch (e) {
       SharedPreferences prefs = await SharedPreferences.getInstance();
-      bool? initScreen = prefs.getBool("initScreenShown");
+      bool? initScreen = prefs.getBool("landingScreenShown"); // landingScreenShown is the boolean value that is used to check wether to show the user the onboarding screen or not on the first launch of the app.
```
Comment (chandansgowda) [inline lib/controllers/auth_state_controller.dart:92]: initScreen should also be renamed to landingScreenShown

File: lib/main.dart [hunk 3]
```diff
@@ -12,7 +12,7 @@ import 'package:shared_preferences/shared_preferences.dart';
 Future<void> main() async {
   WidgetsFlutterBinding.ensureInitialized();
   SharedPreferences prefs = await SharedPreferences.getInstance();
-  bool? initScreen = await prefs.getBool("initScreenShown");
+  bool? initScreen = prefs.getBool("landingScreenShown");
```
Comment (chandansgowda) [inline lib/main.dart:15]: initScreen should also be renamed to landingScreenShown

**Review by chandansgowda — CHANGES_REQUESTED** [review]
@ShivamMenda Please make requested changes and test the flow once.

File: lib/main.dart (hunk 3 above)
Comment (ShivamMenda) [reply lib/main.dart:15]: Sure

File: lib/controllers/auth_state_controller.dart (hunk 2 above)
Comment (ShivamMenda) [reply lib/controllers/auth_state_controller.dart:92]: Sure

**Comment by ShivamMenda** [comment]
@chandansgowda Renaming done. Good to merge.

**Comment by chandansgowda** [comment]
Hey @ShivamMenda 
Can we use GetStorage Instead of Shared Preferences ?

**Comment by ShivamMenda** [comment]
@chandansgowda will implement it

**Comment by ShivamMenda** [comment]
@chandansgowda Implemented with get storage. Good to go.

File: lib/main.dart [hunk 4]
```diff
@@ -7,14 +7,13 @@ import 'package:resonate/routes/app_pages.dart';
...
 import 'package:resonate/utils/ui_sizes.dart';
-import 'package:shared_preferences/shared_preferences.dart';
+import 'package:get_storage/get_storage.dart';
 
 Future<void> main() async {
   WidgetsFlutterBinding.ensureInitialized();
-  SharedPreferences prefs = await SharedPreferences.getInstance();
-  bool? landingScreenShown = prefs.getBool("landingScreenShown");
+  await GetStorage.init();
+  bool? landingScreenShown = GetStorage().read("landingScreenShown");
```
Comment (chandansgowda) [inline lib/main.dart:15]: We don't need to check it in production right? Please remove this line and the log in the next line if not required.

**Review by chandansgowda — CHANGES_REQUESTED** [review]
Small Change. Please look into it.

File: lib/main.dart (hunk 4 above)
Comment (ShivamMenda) [reply lib/main.dart]: @chandansgowda Yes you are right we don't need to check it in prod. It was not necessary, I've removed it. The flow also was tested it works correctly

**Comment by ShivamMenda** [comment]
@chandansgowda Fixed it. The check was not required in prod, so I tested the flow one final time. Everything works.

**Comment by chandansgowda** [comment]
Great Job @ShivamMenda 🚀

**Comment by ShivamMenda** [comment]
@chandansgowda thank you!

**Pull request merged**